x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/mapper/ShapeFieldMapper.java

```diff
@@ -162,7 +163,19 @@ public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext
 
         @Override
         public Query shapeQuery(Geometry shape, String fieldName, ShapeRelation relation, SearchExecutionContext context) {
-            return queryProcessor.shapeQuery(shape, fieldName, relation, context, hasDocValues());
+            failIfNotIndexedNorDocValuesFallback(context);
+            // CONTAINS queries are not supported by VECTOR strategy for indices created before version 7.5.0 (Lucene 8.3.0);
```

Comment:
I know we don't support ES before 7.17, but how far back do we need to support indexed created version?

Reply:
Probably for ever as we offer (I think) the possibility of open old indices read only in a best effort basis.